misc/releaser.py

```diff
@@ -537,7 +598,48 @@ def build_main(args: argparse.Namespace) -> None:
     # TODO: rethink the non-release checks
     # current_version = get_version_from_repo_describe_tag(args.verbose)
     # check_non_release(current_version)
-    build_release(args.version, yes=args.yes)
+    version: Version = args.version
+    yes: bool = args.yes
+    base_ref: str | None = args.base_ref
+
+    if version.is_dev:
+        raise ReleaseError(f"Don't release a dev version: {version}")
+    release_branch = get_release_branch(version)
+    tag = version.with_v_prefix()
+
+    print(f"Building release {COLOR_GREEN}{version}{COLOR_END} ...")
+    print(f"Will use the branch {COLOR_GREEN}{release_branch}{COLOR_END} as a release branch.")
+    print(f"Will annotate the release commit with the tag {COLOR_GREEN}{tag}{COLOR_END}.")
+
+    old_version = ensure_current_code_version_is_smaller(version)
+
+    # Check repo is clean
+    ensure_working_in_a_clean_git_repo()
+
+    ensure_working_on_the_correct_branch(release_branch, base_ref, version)
+
+    release_date, license_eol_date = get_release_date_and_licence_eol()
+
+    # Update BUSL license date marker & version info
+    update_license_file(version, license_eol_date)
+
+    # Update parsec version
+    update_version_files(version)
+
+    # Update HISTORY.rst
```

Comment:
Comments seem redundant because the function names are already well descriptive.

Reply:
done